### ISSUE TITLE
no-explicit-any Broken in typescript paser 5.0.0 and greater

### ISSUE THREAD

**Issue by corbinu** [nzakas/eslint-plugin-typescript#54]
Just a note looks like no-explicit-any has been broken by typescript paser 5.0.0

**Comment by weirdpattern** [comment]
maybe is time to upgrade the parser in our package.json?
Otherwise we will have to fix it in a separate branch...

**Comment by corbinu** [comment]
@weirdpattern Yes that was what I was going to do a PR for. Then saw the errors. I would try to fix but I am at RustConf right now

**Comment by weirdpattern** [comment]
The newer version of the parser removed the node GenericTypeAnnotation in favor of a TSReferenceType for both generics and non-generics...

```ts
/**
         * Checks if the node has a type annotation of type any.
         * @param {ASTNode} node The node being validated.
         * @returns {void}
         * @private
         */
        function checkGenericNodeForAnnotation(node) {
            if (node.type === "TSAnyKeyword") {
                context.report({
                    node,
                    message: "Unexpected any. Specify a different type."
                });
            } else if (node.type === "TSArrayType") {
                checkGenericNodeForAnnotation(node.elementType);
            } else if (node.type === "TSUnionType" || node.type === "TSIntersectionType") {
                node.types.forEach(type => {
                    checkGenericNodeForAnnotation(type);
                });
            } else if (node.type === "TSTypeReference") {
                if (node.typeParameters) { // handles generics
                    node.typeParameters.params.forEach(param => {
                        checkGenericNodeForAnnotation(param);
                    });
                } else if (node.typeName) { // handles non generics
                    checkGenericNodeForAnnotation(node.typeName);
                }
            } else if (node.type === "GenericTypeAnnotation") { 
                if (node.typeParameters) {
                    node.typeParameters.params.forEach(param => {
                        checkGenericNodeForAnnotation(param);
                    });
                } else {
                    checkGenericNodeForAnnotation(node.id);
                }
            }
        }
```

This code should take care of this... we could remove the GenericTypeAnnotation, but I think is best if we leave it there for backwards compatibility...

I've verified both setups work properly... just trying to save you some time, enjoy the conference!!

**Comment by corbinu** [comment]
@weirdpattern Thanks so much opening a PR
